build-scripts/generateServiceClient.ts

```diff
@@ -0,0 +1,197 @@
+
+        console.log('Done generating service client(s)')
+    } finally {
+        // Clean up the temp path
+        del.sync([tempJsSdkPath], { force: true })
+    }
+}
+
+async function cloneJsSdk(destinationPath: string): Promise<void> {
+    console.log('Cloning AWS JS SDK...')
+
+    // Output stderr while it clones so it doesn't look frozen
+    return new Promise<void>((resolve, reject) => {
+        const exec = child_process.exec(
+            `git clone --depth 1 https://github.com/aws/aws-sdk-js.git ${destinationPath}`,
+            {
+                encoding: 'utf8'
+            }
+        )
+
+        exec.stderr.on('data', (data: any) => {
+            console.log(data)
+        })
+
+        exec.once('close', (code, signal) => {
+            exec.stdout.removeAllListeners()
+            resolve()
+        })
+    })
+}
+
+async function insertServiceClientsIntoJsSdk(
+    jsSdkPath: string,
+    serviceClientDefinitions: ServiceClientDefinition[]
+): Promise<void> {
+    serviceClientDefinitions.forEach(serviceClientDefinition => {
+        const apiVersion = getApiVersion(serviceClientDefinition.serviceJsonPath)
+
+        // Copy the Service Json into the JS SDK for generation
+        const jsSdkServiceJsonPath = path.join(
+            jsSdkPath,
+            'apis',
+            `${serviceClientDefinition.serviceName.toLowerCase()}-${apiVersion}.normal.json`
+        )
+        fs.copyFileSync(serviceClientDefinition.serviceJsonPath, jsSdkServiceJsonPath)
+    })
+
+    const apiMetadataPath = path.join(jsSdkPath, 'apis', 'metadata.json')
+    await patchServicesIntoApiMetadata(apiMetadataPath, serviceClientDefinitions.map(x => x.serviceName))
+}
+
+interface ServiceJsonSchema {
+    metadata: {
+        apiVersion: string
+    }
+}
+
+function getApiVersion(serviceJsonPath: string): string {
+    const json = fs.readFileSync(serviceJsonPath).toString()
+    const serviceJson = JSON.parse(json) as ServiceJsonSchema
+
+    return serviceJson.metadata.apiVersion
+}
+
+interface ApiMetadata {
+    [key: string]: { name: string }
+}
+
+/**
+ * Updates the JS SDK's api metadata to contain the provided services
+ */
+async function patchServicesIntoApiMetadata(apiMetadataPath: string, serviceNames: string[]): Promise<void> {
+    console.log(`Patching services (${serviceNames.join(', ')}) into API Metadata...`)
+
+    const apiMetadataJson = fs.readFileSync(apiMetadataPath).toString()
+    const apiMetadata = JSON.parse(apiMetadataJson) as ApiMetadata
+
+    serviceNames.forEach(serviceName => {
+        apiMetadata[serviceName.toLowerCase()] = { name: serviceName }
+    })
+
+    fs.writeFileSync(apiMetadataPath, JSON.stringify(apiMetadata, undefined, 4))
+}
+
+/**
+ * Generates service clients
+ */
+async function runTypingsGenerator(repoPath: string): Promise<void> {
+    console.log('Generating service client typings...')
+
+    const stdout = child_process.execSync('node scripts/typings-generator.js', {
```

Comment:
similar comment about avoiding the shell. Except here it would be `child_process.execFileSync`